### PR TITLE
Deployment, bugfixes and persistent cache

### PR DESCRIPTION
**Pull request by shaygeko**
- Deployment
  - to run locally in dev mode, one now needs to do "npm run dev-start" instead of start
- Some fixes:
    - Removing auto-sign in with google. It wasn't auto-signin in on Danny's machine because of incognito mode
    - Fix for pfp + friend request flooding:
      - Root cause was all the feedItem components checking the cache at the same time and based on that querying the backend. If there was no image in cache at that point, then each feed item sent a request. Also, all images were lost on page reload.
      - (Almost complete) Fix: 
         - Use IndexDB to store the files for long periods of time
         - Introduce loading set - images that have been requested by other components but have not yet arrived. Do not request the profile picture if corresponding user is in the loading set.
         - Set cache eviction timer to 12 hours - don't keep stale data for too long.  
         - This works 4/5 times. In some cases, components still re-request the data, but that is still mitigated by data being persisted between reloads and re-logins
         - I didn't add this logic to userprofile image and the image in nav-header, those arent a big problem if we query for them
     - Infinite load batch size fix in group activity. This still needs to be done in dashboard and user profile pages. Create only one observer - this part might fail the build :D

### CODE REVIEW

File: backend/src/config/sessionConfig.ts [hunk 1]
```diff
@@ -7,8 +7,8 @@ export const sessionOptions = {
     saveUninitialized: false,
     cookie: {
         maxAge: 3600000, // 1 hour in milliseconds
-        // sameSite: 'none' as 'none',
-        // secure: true,
-        // domain: 'codegram-api-vrlbmknooq-uc.a.run.app',
+        sameSite: 'none' as 'none',
+        secure: true,
+        domain: 'backend2-c2h4n4r2vq-uw.a.run.app',
```
Comment (shaygeko) [inline backend/src/config/sessionConfig.ts:12]: we shoulda made this an env variable

File: client/src/components/feed-item/FeedItem.tsx [hunk 2]
```diff
@@ -56,9 +56,14 @@ export const FeedItem = ({
   useEffect(() => {
     const fetchProfilePic = async () => {
       if (username) {
+        if(loadingCache.has(username)){
+          console.log("This pfp is to be loaded by another component");
```
Comment (shaygeko) [inline client/src/components/feed-item/FeedItem.tsx]: oop this should not be in logs

File: client/src/components/friends-list/FriendsList.tsx [hunk 3]
```diff
@@ -15,14 +15,20 @@ interface FriendsListProps {
...
-  const { cache, setCache } = useImageCache();
+  const { cache, setCache, loadingCache, addUsernameToLoadingCache } = useImageCache();
   const [ friendsData, setFriendsData ] = useState<FriendItem[]>([]);
   const showViewAllBtn = sessionUserId === userId;
 
   const fetchProfilePic = useCallback(
     async (username: string) => {
+      
+      if(loadingCache.has(username)){
+        console.log("This pfp is to be loaded by another component");
```
Comment (shaygeko) [inline client/src/components/friends-list/FriendsList.tsx]: same here

File: client/src/pages/user-profile/UserProfile.tsx [hunk 4]
```diff
@@ -140,13 +140,13 @@ export const UserProfilePage = () => {
...
         } else {
-          setProfilePic(currentCache);
+          setProfilePic(currentCache.imageData);
         }
       }
     };
 
     fetchProfilePic();
-  }, [profileUsername, setCache]);
+  }, [profileUsername, cache]);
```
Comment (shaygeko) [inline client/src/pages/user-profile/UserProfile.tsx]: this could be a bug? not sure if it breaks anything

**Review by dannyl1u — DISMISSED** [review]
😅

**Review by dannyl1u — APPROVED** [review]

**Pull request merged**